tests/ha_cases/run.sh

```diff
@@ -96,12 +100,26 @@ function test_join_masters {
     run_dm_master $WORK_DIR/master-join5 $MASTER_PORT5 $cur/conf/dm-master-join5.toml
     check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT5
 
-    run_dm_ctl_with_retry $WORK_DIR 127.0.0.1:$MASTER_PORT1 "query-status" '"result": true' 1
     run_dm_ctl_with_retry $WORK_DIR 127.0.0.1:$MASTER_PORT2 "query-status" '"result": true' 1
     run_dm_ctl_with_retry $WORK_DIR 127.0.0.1:$MASTER_PORT3 "query-status" '"result": true' 1
     run_dm_ctl_with_retry $WORK_DIR 127.0.0.1:$MASTER_PORT4 "query-status" '"result": true' 1
     run_dm_ctl_with_retry $WORK_DIR 127.0.0.1:$MASTER_PORT5 "query-status" '"result": true' 1
-    echo "[$(date)] <<<<<< finish test_join_masters >>>>>>"
+
+    echo "kill dm-master-join1"
+    ps aux | grep dm-master-join1 | awk '{print $2}' | xargs kill || true
+    check_port_offline $MASTER_PORT1 20
+    rm -rf $WORK_DIR/master1/default.master1
+
+    sleep 5
```

Comment:
Is this necessary? Can we test join while transferring leader?

Reply:
That will fail.

Reply:
> That will fail.

what is the error message?

Reply:
`cat /tmp/dm_test/ha_cases/worker1/log/dm-worker.log`


`[2020/05/08 20:43:16.075 +08:00] [INFO] [printer.go:54] ["Welcome to dm-worker"] ["Release Version"=None] ["Git Commit Hash"=None] ["Git Branch"=None] ["UTC Build Time"=None] ["Go Version"=None]
[2020/05/08 20:43:16.075 +08:00] [INFO] [main.go:54] ["dm-worker config"="{\"name\":\"worker1\",\"log-level\":\"info\",\"log-file\":\"/tmp/dm_test/ha_cases/worker1/log/dm-worker.log\",\"log-rotate\":\"\",\"join\":\"127.0.0.1:8261,127.0.0.1:8361,127.0.0.1:8461,127.0.0.1:8561,127.0.0.1:8661\",\"worker-addr\":\"0.0.0.0:8262\",\"advertise-addr\":\"127.0.0.1:8262\",\"config-file\":\"/home/gmhdbjd/code/work/dm/tests/ha_cases/conf/dm-worker-join1.toml\",\"keepalive-ttl\":10}"]
[2020/05/08 20:43:19.078 +08:00] [ERROR] [join.go:52] ["fail to dial dm-master"] [error="context deadline exceeded"]
[2020/05/08 20:43:19.101 +08:00] [ERROR] [join.go:60] ["fail to register worker"] [error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.1:8261: connect: connection refused\""]
[2020/05/08 20:43:19.126 +08:00] [ERROR] [join.go:60] ["fail to register worker"] [error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.1:8261: connect: connection refused\""]
[2020/05/08 20:43:19.138 +08:00] [ERROR] [join.go:60] ["fail to register worker"] [error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.1:8261: connect: connection refused\""]
[2020/05/08 20:43:19.149 +08:00] [ERROR] [join.go:60] ["fail to register worker"] [error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.1:8261: connect: connection refused\""]
[2020/05/08 20:43:19.149 +08:00] [INFO] [main.go:67] ["join the cluster meet error"] [error="cannot connect with master endpoints: [127.0.0.1:8261 127.0.0.1:8361 127.0.0.1:8461 127.0.0.1:8561 127.0.0.1:8661]"] [errorVerbose="cannot connect with master endpoints: [127.0.0.1:8261 127.0.0.1:8361 127.0.0.1:8461 127.0.0.1:8561 127.0.0.1:8661]\ngithub.com/pingcap/dm/dm/worker.(*Server).JoinMaster\n\t/home/gmhdbjd/code/work/dm/dm/worker/join.go:70\ngithub.com/pingcap/dm/cmd/dm-worker.main\n\t/home/gmhdbjd/code/work/dm/cmd/dm-worker/main.go:65\ngithub.com/pingcap/dm/cmd/dm-worker.TestRunMain.func3\n\t/home/gmhdbjd/code/work/dm/cmd/dm-worker/main_test.go:66\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1357"]`

Reply:
This issue is caused by https://github.com/pingcap/dm/blame/master/dm/master/server.go#L269

Reply:
@lichunzhu some of the repeated `transport: Error while dialing dial tcp 127.0.0.1:8261` are returned from DM-master as @GMHDBJD mentioned above.

Reply:
Make sense. @GMHDBJD Excellent job!